crates/hyperswitch_connectors/src/connectors/razorpay/transformers.rs

```diff
@@ -780,219 +237,71 @@ impl<F, T> TryFrom<ResponseRouterData<F, RazorpayPaymentsResponse, T, PaymentsRe
-            net_amount: item.amount,
-            txn_id: connector_request_reference_id.clone(),
-            txn_amount: item.amount,
-            emi_tenure: 0,
-            txn_uuid: uuid::Uuid::new_v4().to_string(),
-            id: ref_id.to_string(),
-            merchant_gateway_account_id: 11476,
-            txn_type: TxnType::AuthAndSettle,
-            redirect: true,
-            version: VERSION,
-            add_to_locker: false,
-            currency: item.router_data.request.currency.to_string(),
-            is_emi: false,
-            gateway: Gateway::Razorpay,
-            date_created: Some(common_utils::date_time::now()),
-            last_modified: Some(common_utils::date_time::now()),
-            ..Default::default()
-        };
-
-        let account_details = AccountDetails {
-            razorpay_id: auth.razorpay_id.clone(),
-            razorpay_secret: auth.razorpay_secret,
-        };
-        let merchant_gateway_account = MerchantGatewayAccount {
-            gateway: Gateway::Razorpay,
-            disabled: false,
-            id: ref_id,
-            account_details: serde_json::to_string(&account_details)
-                .change_context(errors::ConnectorError::ParsingFailed)?,
-            test_mode: false,
-            merchant_id: merchant_data.merchant_id,
-            ..Default::default()
-        };
-        let gateway_txn_data = GatewayTxnData {
-            id: ref_id.to_string(),
-            version: VERSION,
-            gateway_data: "".to_string(),
-            gateway_status: "S".to_string(),
-            match_status: "S".to_string(),
-            txn_detail_id: ref_id.to_string(),
-            date_created: Some(common_utils::date_time::now()),
-            last_updated: Some(common_utils::date_time::now()),
-        };
-        Ok(Self {
-            second_factor,
-            merchant_gateway_account,
-            order_reference,
-            txn_detail,
-            gateway_txn_data,
-        })
-    }
+    let current_time = OffsetDateTime::now_utc().unix_timestamp_nanos();
+    Ok(Some(serde_json::json!(WaitScreenData {
+        display_from_timestamp: current_time,
+        display_to_timestamp: Some(current_time + Duration::minutes(5).whole_nanoseconds()),
+        poll_config: Some(PollConfig {
+            delay_in_secs: 5,
+            frequency: 5,
+        }),
+    })))
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct RazorpaySyncResponse {
-    status: PsyncStatus,
-    is_stateful: bool,
-    second_factor: SecondFactorResponse,
+    id: String,
+    status: RazorpayStatus,
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
-pub enum PsyncStatus {
-    Charged,
-    Pending,
-    Authorizing,
+#[serde(rename_all = "snake_case")]
+#[derive(strum::Display)]
+pub enum RazorpayStatus {
+    Created,
+    Authorized,
+    Captured,
+    Refunded,
+    Failed,
 }
```

Comment:
1. Nit: Can also derive `Copy`.
2. Do note that deriving `strum::Display` will not convert it to string using snake case (because you have `#[serde(rename_all = "snake_case")]`), you'd need a separate annotation / attribute for `strum`. But first of all, why do we need to derive `Display` here, what's the use case?